### PR TITLE
Feature/rollup integration test

### PR DESCRIPTION
**Pull request by mans4singh**
This is the pull request for migrating python roll up tests to tests based on TestNG framework.

### CODE REVIEW

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,1144 @@
...
+			}
+		}
+
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForTomorrowByBucketFromMySQLProdContext()
+			throws Exception {
+
+		Statement statement = connection.createStatement();
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Here you can also use try-with-resource - this Statement and the ResultSet below are never closed.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+				"creationTime", "modificationTime", "ruleJson");
+		experiment.setSerializationStrategy(experimentComparisonStrategy);
+
+	}
+
+	@BeforeClass
+	public void prepareDBConnection() throws Exception {
+		Class.forName("com.mysql.jdbc.Driver").newInstance();
+		connection = DriverManager.getConnection(
+				appProperties.getProperty("database.url"), "readwrite",
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Readwrite is also retrievable from the properties in this PR (see properties file below).

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+		paramsQA.context = QA;
+
+		long startTime = System.currentTimeMillis();
+		ExperimentCounts countsBeforeQA = postExperimentCounts(experiment, paramsQA);
+		long endTime = System.currentTimeMillis();
+
+		System.out.println("Before analytics time QA was: "
+				+ (endTime - startTime));
+
+		startTime = System.currentTimeMillis();
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: 1) Why do we need to measure the times here?
2) This one instance of creating the start time should be a little later, directly before postExperimentCounts is called, to make it more comparable. Also consider using [Instant or nanoTime](http://stackoverflow.com/a/3383047/3004221).

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,1144 @@
...
+				insertIntoRollUp(rollUpQuery);
+			}
+		}
+	}
+
+	private void insertIntoRollUp(String rollUpQuery) throws SQLException {
+		Statement statement = connection.createStatement();
+
+		ResultSet resultSet = statement.executeQuery(rollUpQuery);
+
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Better have a try-with-resource here - if there are problems in the resultSet, the statement might not close this way.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,1144 @@
...
+		Statement statement = connection.createStatement();
+		ResultSet result = statement.executeQuery(query);
+
+		if (!result.next())
+			fail("Should have at least one row for query " + query);
+
+		int userCount = result.getInt("user_count");
+		int distinctUserCount = result.getInt("distinct_user_count");
+
+		result.close();
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: here as well: better use a try-with-resource to wrap the closes into a finally.

File: modules/functional-test/pom.xml [hunk 6]
```diff
@@ -142,5 +142,10 @@
             <artifactId>retrofit1-okhttp3-client</artifactId>
             <version>1.0.2</version>
         </dependency>
+		<dependency>
+		    <groupId>mysql</groupId>
+		    <artifactId>mysql-connector-java</artifactId>
+		    <version>5.1.6</version>
+		</dependency>
```
Comment (Faedrivin) [inline modules/functional-test/pom.xml]: I just annotate it here, but it applies to multiple files: In general we indent with four spaces. One of the files below has 8, one has 2. This one has four, but the dependency should by one level closer to the root.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java (hunk 1 above)
Comment (mans4singh) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Will correct it.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java (hunk 2 above)
Comment (mans4singh) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: You are right.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java (hunk 3 above)
Comment (mans4singh) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: 1. To get an estimate of the processing time before hand.
2. Will correct it.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java (hunk 4 above)
Comment (mans4singh) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Will correct it.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java (hunk 5 above)
Comment (mans4singh) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: Will correct it.

File: modules/functional-test/pom.xml (hunk 6 above)
Comment (mans4singh) [inline modules/functional-test/pom.xml]: Will format it.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,1144 @@
...
+import com.jayway.restassured.response.Response;
+
+import static com.intuit.wasabi.tests.library.util.Constants.*;
+
+/**
+ * RollUp integration tests
+ */
+public class RollUpIntegrationTest extends TestBase {
+
+	private static final String CONTEXT = "context";
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java]: This class seems to be using an indentation of 7 spaces.

File: modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpBadCallsIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+import com.intuit.wasabi.tests.model.factory.UserFactory;
+import com.jayway.restassured.response.Response;
+
+import static com.intuit.wasabi.tests.library.util.Constants.*;
+/**
+ * Bucket integration tests
+ */
+public class RollUpBadCallsIntegrationTest extends TestBase {
+
+	private static final String FROM_TIME = "fromTime";
```
Comment (Faedrivin) [inline modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpBadCallsIntegrationTest.java]: Here the indentation is 7 spaces, further below correct with 4 but even further with only 2.

**Comment by Faedrivin** [comment]
Apart from some indentations which can be resolved during the merge this is great, so :shipit:

**Comment by mans4singh** [comment]
Formatted tests

**Pull request merged**